### PR TITLE
Replace many conversion functions with `From` impls.

### PR DESCRIPTION
**Pull request by sunfishcode**
Replace many of the `conv` functions for converting types into `ArgReg`
values with `impl From<T> for ArgReg` impls. This requires a little more
boilerplate in the inline asm code to call `.into()` on all the
arguments, but it makes the main syscall code much less verbose.

### CODE REVIEW

File: src/imp/linux_raw/arch/inline/aarch64.rs [hunk 1]
```diff
@@ -21,259 +21,265 @@ pub(in crate::imp) unsafe fn syscall0_readonly(nr: SyscallNumber<'_>) -> RetReg<
...
 #[must_use]
-pub(in crate::imp) unsafe fn syscall3_readonly(
+pub(in crate::imp) unsafe fn syscall3_readonly<'a>(
     nr: SyscallNumber<'_>,
-    a0: ArgReg<'_, A0>,
-    a1: ArgReg<'_, A1>,
-    a2: ArgReg<'_, A2>,
+    a0: impl Into<ArgReg<'a, A0>>,
+    a1: impl Into<ArgReg<'a, A1>>,
+    a2: impl Into<ArgReg<'a, A2>>,
```
Comment (cgwalters) [inline src/imp/linux_raw/arch/inline/aarch64.rs:130]: Any concerns about the implicit generics here potentially introducing multiple copies of this and other functions?  Might we need to do the "manual forward to non-generic impl" trick?  (Seems like that'd be painful here though...)

Or maybe the inline asm is small enough this doesn't matter.

**Review by cgwalters — APPROVED** [review]
I only gave this a very superficial skim...I think it seems sane.

File: src/imp/linux_raw/arch/inline/aarch64.rs (hunk 1 above)
Comment (sunfishcode) [reply src/imp/linux_raw/arch/inline/aarch64.rs:130]: Ah, that's a good point. In release builds these I expect these to be inlined. But I should look more closely at what happens in practice, and also what happens in debug builds.

Comment (cgwalters) [reply src/imp/linux_raw/arch/inline/aarch64.rs:130]: Probably simplest to `#[inline(always)]`?

Comment (sunfishcode) [reply src/imp/linux_raw/arch/inline/aarch64.rs]: It's a good idea, but thinking about this more, I think this is a good use case for a few macros. They can help factor out even more of the redundancy, including a bunch of `#cfg`'s in the syscalls.rs files, which is quite nice and appears enough for me to set aside my general dislike of macros in this case :-).

**Pull request merged**